### PR TITLE
feat(gatsby-source-wordpress): Add keepMediaDetails

### LINKED ISSUE #6509: [gatsby-source-wordpress] Wordpress Media Sizes
## Description

I am using https://wordpress.org/plugins/manual-image-crop/ for finer controls on cropping for different image sizes/ratios that my application is using.

However, the `media_details.sizes` info is currently being stripped away by the plugin in favor of localFile.

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-wordpress/src/normalize.js#L457

### Steps to reproduce

Create any post with image custom field and try to query it on GraphiQL

### Expected result

`$imagePath.media_details.sizes` should return the different sizes for the image files

Ideally, we should be able to transform the different sizes with `gatsby-transformer-sharp` as well.

### Actual result

`$imagePath.media_details.sizes` is set to null

### Environment

```
 System:
    OS: macOS High Sierra 10.13.5
    CPU: x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.4.1 - ~/.nvm/versions/node/v10.4.1/bin/node
    Yarn: 1.7.0 - /usr/local/bin/yarn
    npm: 6.1.0 - ~/.nvm/versions/node/v10.4.1/bin/npm
  Browsers:
    Chrome: 67.0.3396.99
    Firefox: 60.0.1
    Safari: 11.1.1
  npmPackages:
    gatsby: next => 2.0.0-beta.42
    gatsby-image: next => 2.0.0-beta.6
    gatsby-plugin-glamor: next => 2.0.0-beta.2
    gatsby-plugin-react-helmet: next => 3.0.0-beta.3
    gatsby-plugin-sharp: next => 2.0.0-beta.5
    gatsby-plugin-typography: next => 2.2.0-beta.2
    gatsby-source-wordpress: next => 3.0.0-beta.7
    gatsby-transformer-sharp: next => 2.1.1-beta.4
```

### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

### PR DESCRIPTION
**Pull request by billyjacoby**
## Description
This will enable users to toggle sourceThumbnails in order to decide whether or not thumbnail info is downloaded.

I needed this in order to build an image carousel that didn't warrant the highest quality photos to speed up site load times. 

## Related Issues
Fixes #6509

### CODE REVIEW

**Comment by amberleyromo** [comment]
@gatsbyjs/learning 	is tagged on this, but more appropriate for @gatsbyjs/core 	-- will leave for y'all!

**Comment by sidharthachatterjee** [comment]
Closing this for now since the changes are hard to grok with pushing untranspiled code 

Let's re open after pushing untranspiled code and take it from there!

**Comment by billyjacoby** [comment]
I just committed the changes to my fork, sorry about that!

**Comment by billyjacoby** [comment]
> You seem to have pushed built files in `src/gatsby-node.js`
> 
> Could you please push the src (without transpilation)? Also, it would be nice to add a test for this new option.

This is done!

**Comment by sidharthachatterjee** [comment]
@billyjacoby No problem. Thanks for fixing! Could you please fix conflicts and add a test for the new option! Otherwise, looks good! 🙌🏼

**Comment by billyjacoby** [comment]
> @billyjacoby No problem. Thanks for fixing! Could you please fix conflicts and add a test for the new option! Otherwise, looks good! 🙌🏼

Conflicts are resolved.

As for adding tests, "it's a bit tricky to mock as it needs to access to the store/cache + would download file.." according to the existing code base. To be honest I'm not sure how I'd go about mocking all of that...

**Comment by sidharthachatterjee** [comment]
`gatsby-source-wordpress/src/normalize.js` still appears to be transpiled code 😞 

Could you please push the source so I can review this? 

> As for adding tests, "it's a bit tricky to mock as it needs to access to the store/cache + would download file.." according to the existing code base. To be honest I'm not sure how I'd go about mocking all of that...

Just took a look. Fair enough. Since this is fairly simple (only 'keeps' some keys that would otherwise be deleted) I think we can skip testing this till we figure out how to better mock the download and store etc

File: packages/gatsby-source-wordpress/README.md [hunk 1]
```diff
@@ -135,6 +135,9 @@ module.exports = {
         ],
         // Blacklisted routes using glob patterns
         excludedRoutes: ["**/posts/1456"],
+        // Set whether or not thumbnail URLS are sourced.
+        // Defaults to false.
+        sourceThumbnails: false,
```
Comment (sidharthachatterjee) [inline packages/gatsby-source-wordpress/README.md]: Let's call this `keepMediaDetails` to make obvious what this option does

It isn't immediately obvious what `sourceThumbnails` does without enough context

File: packages/gatsby-source-wordpress/src/normalize.js [hunk 2]
```diff
@@ -1,141 +1,168 @@
+"use strict"
```
Comment (sidharthachatterjee) [inline packages/gatsby-source-wordpress/src/normalize.js]: Need to revert to source

**Comment by billyjacoby** [comment]
> `gatsby-source-wordpress/src/normalize.js` still appears to be transpiled code 😞
> 
> Could you please push the source so I can review this?
> 
> > As for adding tests, "it's a bit tricky to mock as it needs to access to the store/cache + would download file.." according to the existing code base. To be honest I'm not sure how I'd go about mocking all of that...
> 
> Just took a look. Fair enough. Since this is fairly simple (only 'keeps' some keys that would otherwise be deleted) I think we can skip testing this till we figure out how to better mock the download and store etc

I totally just forgot to push after making the changes locally last night. Should be good to go now.

File: packages/gatsby-source-wordpress/src/normalize.js (hunk 2 above)
Comment (billyjacoby) [reply packages/gatsby-source-wordpress/src/normalize.js]: Should be reverted with latest commit. Sorry!

**Comment by sidharthachatterjee** [comment]
@billyjacoby Thanks! What do you think about renaming the option?

**Comment by billyjacoby** [comment]
> @billyjacoby Thanks! What do you think about renaming the option?

I'm totally fine with it. That was just the most concise name I could think of, but naming is certainly not my strong suite😅 ...

**Comment by billyjacoby** [comment]
> @billyjacoby Thanks! What do you think about renaming the option?

Maybe just 'thumbnailURLs'?

**Comment by sidharthachatterjee** [comment]
@billyjacoby Oops, I hit the wrong button hah

No worries. Naming is hard! Go ahead and make the change then and we'll get this shipped 💃

File: packages/gatsby-source-wordpress/README.md (hunk 1 above)
Comment (billyjacoby) [reply packages/gatsby-source-wordpress/README.md]: Just saw this. Did you make the changes or should i?

Comment (sidharthachatterjee) [reply packages/gatsby-source-wordpress/README.md]: I haven't made them yet but I can if you like!

**Comment by billyjacoby** [comment]
> @billyjacoby Oops, I hit the wrong button hah
> 
> No worries. Naming is hard! Go ahead and make the change then and we'll get this shipped 💃

Done! I liked your name better than mine, so thanks for that! 😬

**Comment by sidharthachatterjee** [comment]
@billyjacoby Can you please give me push access to your fork? Just did a little clean up and tried to push

File: packages/gatsby-source-wordpress/src/gatsby-node.js [hunk 3]
```diff
@@ -138,7 +142,7 @@ exports.sourceNodes = async (
     touchNode,
     getNode,
     _auth,
-    reporter,
```
Comment (pieh) [inline packages/gatsby-source-wordpress/src/gatsby-node.js]: We still should pass `reporter` to `normalize.downloadMediaFiles`

Comment (sidharthachatterjee) [reply packages/gatsby-source-wordpress/src/gatsby-node.js]: Fixing that in https://github.com/gatsbyjs/gatsby/pull/16670

Not published yet anyway

**Comment by sidharthachatterjee** [comment]
Renamed this to `keepMediaSizes` in https://github.com/gatsbyjs/gatsby/pull/16670 because I realised we were deleting `sizes` and not `media_details`

Sorry for the back and forth @billyjacoby

**Comment by sidharthachatterjee** [comment]
Published in `gatsby-source-wordpress@3.1.16`

**Pull request merged**